server/src/main/java/com/cloud/hypervisor/KVMGuru.java

```diff
@@ -213,7 +213,7 @@ protected void configureVmMemoryAndCpuCores(VirtualMachineTO virtualMachineTo, H
         Integer maxHostCpuCore = max.second();
 
         long minMemory = virtualMachineTo.getMinRam();
-        Long maxMemory = minMemory;
+        Long maxMemory = virtualMachineTo.getMaxRam();
```

Comment:
I think only this fix was enough :)

Reply:
Closing my PR as this main line/change has been addressed - https://github.com/apache/cloudstack/pull/7807